### PR TITLE
Feature/support for password file

### PR DESCRIPTION
**Pull request by aussiDavid**
Add support for PASSWORD_FILE environment for the docker image

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -19,6 +18,23 @@ HIDE_SHARES="${HIDE_SHARES:-no}"
 
 # common functions
 set_password() {
+  # check PASSWORD and PASSWORD_FILE are both not set
```
Comment (mbentley) [inline entrypoint.sh]: I would simplify this and replace `! -z` with `-n`:
```
# check PASSWORD and PASSWORD_FILE are both not set
if [ -n "${PASSWORD}" ] && [ -n "${PASSWORD_FILE}" ]
then
  echo "ERROR: PASSWORD and PASSWORD_FILE can not both be set. Please choose 1"
  exit 1
fi
```

File: entrypoint.sh [hunk 2]
```diff
@@ -19,6 +18,23 @@ HIDE_SHARES="${HIDE_SHARES:-no}"
...
+  if [ ! -z "${PASSWORD}" ]
+  then
+    if [ ! -z "${PASSWORD_FILE}" ]
+    then
+      echo "ERROR: PASSSWORD and PASSWORD_FILE can not both be set. Please choose 1"
```
Comment (mbentley) [inline entrypoint.sh]: `PASSSWORD` -> `PASSWORD`

File: entrypoint.sh [hunk 3]
```diff
@@ -19,6 +18,23 @@ HIDE_SHARES="${HIDE_SHARES:-no}"
...
+      exit 1
+    fi
+  fi
+
+  PASSWORD="${PASSWORD:-timemachine}"
+
+  if [ ! -z "${PASSWORD_FILE}" ]
```
Comment (mbentley) [inline entrypoint.sh]: replace `! -z` with `-n`

File: entrypoint.sh [hunk 4]
```diff
@@ -19,6 +18,23 @@ HIDE_SHARES="${HIDE_SHARES:-no}"
...
+  then
+    PASSWORD="$(cat $PASSWORD_FILE)"
```
Comment (mbentley) [inline entrypoint.sh]: change `cat $PASSWORD_FILE` to `cat "${PASSWORD_FILE}"` to prevent globbing and word splitting + add `{` `}` for style preference.

Comment (aussiDavid) [reply entrypoint.sh]: Done. Thank you

File: entrypoint.sh (hunk 3 above)
Comment (aussiDavid) [reply entrypoint.sh]: Done. Thank you

File: entrypoint.sh (hunk 2 above)
Comment (aussiDavid) [reply entrypoint.sh]: Done. Thank you

File: entrypoint.sh (hunk 1 above)
Comment (aussiDavid) [reply entrypoint.sh]: Done. Thank you

**Comment by aussiDavid** [comment]
Glad to help :)

**Pull request merged**